test_version_update_after_mutation can't start server after restart

test_version_update_after_mutation/test.py::test_mutate_and_upgrade

https://s3.amazonaws.com/clickhouse-test-reports/44897/2e758f983fb86732e0467f6a9bc2deb266d51a40/integration_tests__release__[4/4].html

```
start_cluster = <helpers.cluster.ClickHouseCluster object at 0x7fae3b17cf40>

    def test_mutate_and_upgrade(start_cluster):
        for node in [node1, node2]:
            node.query("DROP TABLE IF EXISTS mt")
            node.query(
                "CREATE TABLE mt (EventDate Date, id UInt64) ENGINE ReplicatedMergeTree('/clickhouse/tables/t', '{}') ORDER BY tuple()".format(
                    node.name
                )
            )
    
        node1.query("INSERT INTO mt VALUES ('2020-02-13', 1), ('2020-02-13', 2);")
    
        node1.query("ALTER TABLE mt DELETE WHERE id = 2", settings={"mutations_sync": "2"})
        node2.query("SYSTEM SYNC REPLICA mt", timeout=15)
    
        node2.query("DETACH TABLE mt")  # stop being leader
        node1.query("DETACH TABLE mt")  # stop being leader
        node1.restart_with_latest_version(signal=9, fix_metadata=True)
>       node2.restart_with_latest_version(signal=9, fix_metadata=True)

test_version_update_after_mutation/test.py:62: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
helpers/cluster.py:3697: in restart_with_latest_version
    self.wait_start(time_left)
```

```
2023.01.04 21:17:22.151236 [ 142 ] {} <Error> Application: Caught exception while loading metadata: Code: 219. DB::Exception: Cannot drop: filesystem error: in remove: Directory not empty ["/var/lib/clickhouse/data
/system/"]. Probably database contain some detached tables or metadata leftovers from Ordinary engine. If you want to remove all data anyway, try to attach database back and drop it again with enabled force_remove_
data_recursively_on_drop setting: Exception while trying to convert database system from Ordinary to Atomic. It may be in some intermediate state. You can finish conversion manually by moving the rest tables from s
ystem to .tmp_convert.system.16495121816668103290 (using RENAME TABLE) and executing DROP DATABASE system and RENAME DATABASE .tmp_convert.system.16495121816668103290 TO system. (DATABASE_NOT_EMPTY), Stack trace (w
hen copying this message, always include the lines below):

0. DB::Exception::Exception(DB::Exception::MessageMasked const&, int, bool) @ 0xf2d0b7a in /usr/bin/clickhouse
1. ? @ 0xde1842e in /usr/bin/clickhouse
2. DB::DatabaseOnDisk::drop(std::__1::shared_ptr<DB::Context const>) @ 0x13dc7e1f in /usr/bin/clickhouse
3. DB::DatabaseCatalog::detachDatabase(std::__1::shared_ptr<DB::Context const>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, bool, bool) @ 0x140687f4 in /usr/bin/clic
khouse
4. DB::InterpreterDropQuery::executeToDatabaseImpl(DB::ASTDropQuery const&, std::__1::shared_ptr<DB::IDatabase>&, std::__1::vector<StrongTypedef<wide::integer<128ul, unsigned int>, DB::UUIDTag>, std::__1::allocator
<StrongTypedef<wide::integer<128ul, unsigned int>, DB::UUIDTag>>>&) @ 0x148657eb in /usr/bin/clickhouse
5. DB::InterpreterDropQuery::executeToDatabase(DB::ASTDropQuery const&) @ 0x148621bd in /usr/bin/clickhouse
6. DB::InterpreterDropQuery::execute() @ 0x1486155c in /usr/bin/clickhouse
7. ? @ 0x14d417b3 in /usr/bin/clickhouse
8. DB::executeQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::shared_ptr<DB::Context>, bool, DB::QueryProcessingStage::Enum) @ 0x14d3e86d in /usr/bin/cli
ckhouse
9. ? @ 0x14d657fe in /usr/bin/clickhouse
10. DB::maybeConvertSystemDatabase(std::__1::shared_ptr<DB::Context>) @ 0x14d6372f in /usr/bin/clickhouse
11. DB::Server::main(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::al
locator<char>>>> const&) @ 0xf360de1 in /usr/bin/clickhouse
12. Poco::Util::Application::run() @ 0x188bb662 in /usr/bin/clickhouse
13. DB::Server::run() @ 0xf350d79 in /usr/bin/clickhouse
14. Poco::Util::ServerApplication::run(int, char**) @ 0x188cf9ba in /usr/bin/clickhouse
15. mainEntryClickHouseServer(int, char**) @ 0xf34d7de in /usr/bin/clickhouse
16. main @ 0x89c1f73 in /usr/bin/clickhouse
17. __libc_start_main @ 0x21b97 in /lib/x86_64-linux-gnu/libc-2.27.so
18. _start @ 0x7f39a2e in /usr/bin/clickhouse
 (version 22.13.1.1257)
2023.01.04 21:17:22.202766 [ 142 ] {} <Information> Application: Shutting down storages.
```

Another similar failure: https://s3.amazonaws.com/clickhouse-test-reports/42181/85734bb321db4561e0b4e871535954588bb96c27/integration_tests__tsan__[2/6].html

It's much more interesting that there's sanitizer assert:
```
E           Exception: Sanitizer assert found in /ClickHouse/tests/integration/test_version_update_after_mutation/_instances_0/docker.log for instance ==================
E           ==================
E           ==================
E           ==================
```
node3/logs/stderr.log:
```
==142==WARNING: invalid path to external symbolizer!
==142==WARNING: Failed to use and restart external symbolizer!
==================
WARNING: ThreadSanitizer: data race (pid=142)
  Write of size 8 at 0x7b7800bb0b58 by main thread (mutexes: write M0, write M1):
    #0 <null> <null> (clickhouse+0xc14f26b) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #1 <null> <null> (clickhouse+0x1e96a14b) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #2 <null> <null> (clickhouse+0x1c78d386) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #3 <null> <null> (clickhouse+0x1d7d1cca) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #4 <null> <null> (clickhouse+0x1d7d0854) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #5 <null> <null> (clickhouse+0x1dc8ae85) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #6 <null> <null> (clickhouse+0x1dc8732b) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #7 <null> <null> (clickhouse+0x1dcb51c6) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #8 <null> <null> (clickhouse+0x1dcb3632) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #9 <null> <null> (clickhouse+0x144ab8e3) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #10 <null> <null> (clickhouse+0x22a44dbd) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #11 <null> <null> (clickhouse+0x1449b57a) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #12 <null> <null> (clickhouse+0x22a64c6f) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #13 <null> <null> (clickhouse+0x14498154) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #14 <null> <null> (clickhouse+0xc135ad1) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #15 <null> <null> (libc.so.6+0x21b96) (BuildId: d3cf764b2f97ac3efe366ddd07ad902fb6928fd7)

  Previous read of size 8 at 0x7b7800bb0b58 by thread T70 (mutexes: write M2):
    #0 <null> <null> (clickhouse+0x1e94f9bf) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #1 <null> <null> (clickhouse+0x1e95b6b8) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #2 <null> <null> (clickhouse+0x1e9d6506) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #3 <null> <null> (clickhouse+0x1c28bb1e) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #4 <null> <null> (clickhouse+0x1c28e7d2) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #5 <null> <null> (clickhouse+0x1c28f245) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #6 <null> <null> (clickhouse+0x144f2e10) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #7 <null> <null> (clickhouse+0x144f7611) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #8 <null> <null> (clickhouse+0xc0b4326) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)

  Location is heap block of size 2928 at 0x7b7800bb0800 allocated by thread T195:
    #0 <null> <null> (clickhouse+0xc133d06) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #1 <null> <null> (clickhouse+0x1ecb4755) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #2 <null> <null> (clickhouse+0x1ecb0936) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #3 <null> <null> (clickhouse+0x1bf01cc2) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #4 <null> <null> (clickhouse+0x1e3f9f29) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #5 <null> <null> (clickhouse+0x1c7871d1) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #6 <null> <null> (clickhouse+0x1c7a9ef0) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #7 <null> <null> (clickhouse+0x1c809b4b) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #8 <null> <null> (clickhouse+0x144f5bb0) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #9 <null> <null> (clickhouse+0x144fa0d7) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #10 <null> <null> (clickhouse+0x144fa041) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #11 <null> <null> (clickhouse+0x144f2e10) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #12 <null> <null> (clickhouse+0x144f7611) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #13 <null> <null> (clickhouse+0xc0b4326) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)

  Mutex M0 (0x7b0c001501e0) created at:
    #0 <null> <null> (clickhouse+0xc0d209a) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #1 <null> <null> (clickhouse+0x26685a79) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #2 <null> <null> (clickhouse+0x1cad1325) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #3 <null> <null> (clickhouse+0x1cac6874) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #4 <null> <null> (clickhouse+0x1d7d079d) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #5 <null> <null> (clickhouse+0x1dc8ae85) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #6 <null> <null> (clickhouse+0x1dc8732b) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #7 <null> <null> (clickhouse+0x1dcb51c6) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #8 <null> <null> (clickhouse+0x1dcb3632) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #9 <null> <null> (clickhouse+0x144ab8e3) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #10 <null> <null> (clickhouse+0x22a44dbd) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #11 <null> <null> (clickhouse+0x1449b57a) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #12 <null> <null> (clickhouse+0x22a64c6f) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #13 <null> <null> (clickhouse+0x14498154) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #14 <null> <null> (clickhouse+0xc135ad1) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #15 <null> <null> (libc.so.6+0x21b96) (BuildId: d3cf764b2f97ac3efe366ddd07ad902fb6928fd7)

  Mutex M1 (0x7b0c00150360) created at:
    #0 <null> <null> (clickhouse+0xc0d209a) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #1 <null> <null> (clickhouse+0x26685a79) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #2 <null> <null> (clickhouse+0x1cad1325) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #3 <null> <null> (clickhouse+0x1cac6874) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #4 <null> <null> (clickhouse+0x1d7d079d) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #5 <null> <null> (clickhouse+0x1dc8ae85) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #6 <null> <null> (clickhouse+0x1dc8732b) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #7 <null> <null> (clickhouse+0x1dcb51c6) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #8 <null> <null> (clickhouse+0x1dcb3632) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #9 <null> <null> (clickhouse+0x144ab8e3) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #10 <null> <null> (clickhouse+0x22a44dbd) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #11 <null> <null> (clickhouse+0x1449b57a) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #12 <null> <null> (clickhouse+0x22a64c6f) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #13 <null> <null> (clickhouse+0x14498154) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #14 <null> <null> (clickhouse+0xc135ad1) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #15 <null> <null> (libc.so.6+0x21b96) (BuildId: d3cf764b2f97ac3efe366ddd07ad902fb6928fd7)

  Mutex M2 (0x7b3400012d08) created at:
    #0 <null> <null> (clickhouse+0xc0d209a) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #1 <null> <null> (clickhouse+0x26685a79) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #2 <null> <null> (clickhouse+0x1c28ba9e) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #3 <null> <null> (clickhouse+0x1c28e7d2) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #4 <null> <null> (clickhouse+0x1c28f245) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #5 <null> <null> (clickhouse+0x144f2e10) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #6 <null> <null> (clickhouse+0x144f7611) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #7 <null> <null> (clickhouse+0xc0b4326) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)

  Thread T70 'BgSchPool' (tid=226, running) created by main thread at:
    #0 <null> <null> (clickhouse+0xc0b43bd) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #1 <null> <null> (clickhouse+0x144f6d9b) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #2 <null> <null> (clickhouse+0x144f147d) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #3 <null> <null> (clickhouse+0x144f2052) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #4 <null> <null> (clickhouse+0x1c28ca11) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #5 <null> <null> (clickhouse+0x1ca1c32b) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #6 <null> <null> (clickhouse+0x1cabbf85) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #7 <null> <null> (clickhouse+0x144ab2c8) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #8 <null> <null> (clickhouse+0x22a44dbd) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #9 <null> <null> (clickhouse+0x1449b57a) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #10 <null> <null> (clickhouse+0x22a64c6f) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #11 <null> <null> (clickhouse+0x14498154) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #12 <null> <null> (clickhouse+0xc135ad1) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #13 <null> <null> (libc.so.6+0x21b96) (BuildId: d3cf764b2f97ac3efe366ddd07ad902fb6928fd7)

  Thread T195 'ThreadPool' (tid=351, running) created by main thread at:
    #0 <null> <null> (clickhouse+0xc0b43bd) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #1 <null> <null> (clickhouse+0x144f6d9b) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #2 <null> <null> (clickhouse+0x144f147d) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #3 <null> <null> (clickhouse+0x144f97a5) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #4 <null> <null> (clickhouse+0x144f3e79) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #5 <null> <null> (clickhouse+0x144f3aa5) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #6 <null> <null> (clickhouse+0x1c791089) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #7 <null> <null> (clickhouse+0x1c7a9972) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #8 <null> <null> (clickhouse+0x1c8070f1) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #9 <null> <null> (clickhouse+0x1dcb91af) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #10 <null> <null> (clickhouse+0x144ab89f) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #11 <null> <null> (clickhouse+0x22a44dbd) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #12 <null> <null> (clickhouse+0x1449b57a) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #13 <null> <null> (clickhouse+0x22a64c6f) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #14 <null> <null> (clickhouse+0x14498154) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #15 <null> <null> (clickhouse+0xc135ad1) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7)
    #16 <null> <null> (libc.so.6+0x21b96) (BuildId: d3cf764b2f97ac3efe366ddd07ad902fb6928fd7)

SUMMARY: ThreadSanitizer: data race (/usr/bin/clickhouse+0xc14f26b) (BuildId: 1f15bee3bcdffee92b5538ed9c1c2dbedaea5cf7) 
```

--- a/tests/integration/helpers/cluster.py
+++ b/tests/integration/helpers/cluster.py
@@ -3199,7 +3199,6 @@ class ClickHouseInstance:
     ):
         self.name = name
         self.base_cmd = cluster.base_cmd
-        self.base_dir = base_path
         self.docker_id = cluster.get_instance_docker_id(self.name)
         self.cluster = cluster
         self.hostname = hostname if hostname is not None else self.name
@@ -4193,14 +4192,6 @@ class ClickHouseInstance:
         self.exec_in_container(
             ["bash", "-c", f"sed -i 's/{replace}/{replacement}/g' {path_to_config}"]
         )
-
-    def put_users_config(self, config_path):
-        """Put new config (useful if you cannot put it at the start)"""
-
-        instance_config_dir = p.abspath(p.join(self.path, "configs"))
-        users_d_dir = p.abspath(p.join(instance_config_dir, "users.d"))
-        config_path = p.join(self.base_dir, config_path)
-        shutil.copy(config_path, users_d_dir)
 
     def create_dir(self):
         """Create the instance directory and all the needed files there."""

--- a/tests/integration/test_version_update_after_mutation/configs/force_remove_data_recursively_on_drop.xml
+++ b/tests/integration/test_version_update_after_mutation/configs/force_remove_data_recursively_on_drop.xml
@@ -1,7 +1,0 @@
-<clickhouse>
-    <profiles>
-        <default>
-            <force_remove_data_recursively_on_drop>1</force_remove_data_recursively_on_drop>
-        </default>
-    </profiles>
-</clickhouse>

--- a/tests/integration/test_version_update_after_mutation/test.py
+++ b/tests/integration/test_version_update_after_mutation/test.py
@@ -51,12 +51,6 @@ def start_cluster():
         cluster.shutdown()
 
 
-def restart_node(node):
-    # set force_remove_data_recursively_on_drop (cannot be done before, because the version is too old)
-    node.put_users_config("configs/force_remove_data_recursively_on_drop.xml")
-    node.restart_with_latest_version(signal=9, fix_metadata=True)
-
-
 def test_mutate_and_upgrade(start_cluster):
     for node in [node1, node2]:
         node.query("DROP TABLE IF EXISTS mt")
@@ -73,9 +67,8 @@ def test_mutate_and_upgrade(start_cluster):
 
     node2.query("DETACH TABLE mt")  # stop being leader
     node1.query("DETACH TABLE mt")  # stop being leader
-
-    restart_node(node1)
-    restart_node(node2)
+    node1.restart_with_latest_version(signal=9, fix_metadata=True)
+    node2.restart_with_latest_version(signal=9, fix_metadata=True)
 
     # After hard restart table can be in readonly mode
     exec_query_with_retry(
@@ -131,7 +124,7 @@ def test_upgrade_while_mutation(start_cluster):
     # (We could be in process of creating some system table, which will leave empty directory on restart,
     # so when we start moving system tables from ordinary to atomic db, it will complain about some undeleted files)
     node3.query("SYSTEM FLUSH LOGS")
-    restart_node(node3)
+    node3.restart_with_latest_version(signal=9, fix_metadata=True)
 
     # checks for readonly
     exec_query_with_retry(node3, "OPTIMIZE TABLE mt1", sleep_time=5, retry_count=60)

--- a/tests/integration/test_version_update_after_mutation/test.py
+++ b/tests/integration/test_version_update_after_mutation/test.py
@@ -67,6 +67,8 @@ def test_mutate_and_upgrade(start_cluster):
 
     node2.query("DETACH TABLE mt")  # stop being leader
     node1.query("DETACH TABLE mt")  # stop being leader
+    node1.query("SYSTEM FLUSH LOGS")
+    node2.query("SYSTEM FLUSH LOGS")
     node1.restart_with_latest_version(signal=9, fix_metadata=True)
     node2.restart_with_latest_version(signal=9, fix_metadata=True)
 

